superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterBar.tsx

```diff
@@ -183,14 +177,10 @@ const FilterBar: React.FC<FiltersBarProps> = ({
   directPathToChild,
 }) => {
   const [filterData, setFilterData] = useState<{
-    [id: string]: {
-      extraFormData: ExtraFormData;
-      currentState: CurrentFilterState;
-    };
+    [filterId: string]: Omit<FilterState, 'id'>;
```

Comment:
..this would also be cleaner with `id` being optional.

Reply:
I think same thing :)